### PR TITLE
[master] RHCOS image import support

### PR DESCRIPTION
**Pull request by PanYaLian**
Update the doc change to support rhcos image
Add disk_type for RHCOS image import
image export also returns comments info

Signed-off-by: ylpan <ylpan@cn.ibm.com>

### CODE REVIEW

File: doc/source/parameters.yaml [hunk 1]
```diff
@@ -909,6 +910,14 @@ guest_os_version:
   in: body
   required: true
   type: string
+guest_os_version_all:
+  description: |
+    Operating system version, the valid os_version values are: rhel6.x, rhel7.x,
```
Comment (jichenjc) [inline doc/source/parameters.yaml]: @PanYaLian can you help to talk to @dongyanyang about rhel8 support?

File: zvmsdk/smtclient.py [hunk 2]
```diff
@@ -2149,6 +2173,30 @@ def _get_disk_size_units(self, image_path):
         LOG.debug("The image's root_disk_units is %s" % root_disk_units)
         return root_disk_units
 
+    def _get_disk_size_units_rhcos(self, image_path):
```
Comment (jichenjc) [inline zvmsdk/smtclient.py]: can I know the difference between rhcos and normal OS?
you calculate the OS size by using disk size?

File: doc/source/parameters.yaml (hunk 1 above)
Comment (PanYaLian) [reply doc/source/parameters.yaml]: have talked with dongyan, they will include rhel8 doc change in other PR

File: zvmsdk/smtclient.py (hunk 2 above)
Comment (PanYaLian) [reply zvmsdk/smtclient.py]: have rewrite the code to use fdisk to get size

File: zvmsdk/smtclient.py [hunk 3]
```diff
@@ -2062,15 +2079,21 @@ def image_import(self, image_name, url, image_meta, remote_host=None):
 
             # TODO: put multiple disk image into consideration, update the
             # disk_size_units and image_size db field
-            disk_size_units = self._get_disk_size_units(final_image_fpath)
+            if not image_os_version.lower().startswith('rhcos'):
```
Comment (jichenjc) [inline zvmsdk/smtclient.py]: I suggest to make 
image_os_version.lower().startswith('rhcos'):
as a function such as 'is_rhcos‘
then you can reuse that function everywhere

File: zvmsdk/smtclient.py [hunk 4]
```diff
@@ -2163,6 +2188,45 @@ def _get_disk_size_units(self, image_path):
...
+                                             stderr=subprocess.STDOUT)
+        except subprocess.CalledProcessError as err:
+            rc = err.returncode
+            output = err.output
+        except Exception as err:
+            err_msg = ('Command "%s" Error: %s' % (' '.join(command),
+                                                   str(err)))
+            raise exception.SDKInternalError(msg=err_msg)
+
+        if rc or output.strip('1234567890\n'):
```
Comment (jichenjc) [inline zvmsdk/smtclient.py]: what's this? output failed output.strip('1234567890\n'):?

Comment (PanYaLian) [reply zvmsdk/smtclient.py]: The result should be the bytes of os file systems, only number and '\n' should be include, if there are othere characters, it means error happened

File: zvmsdk/smtclient.py (hunk 3 above)
Comment (PanYaLian) [reply zvmsdk/smtclient.py]: done

File: zvmsdk/smtclient.py [hunk 5]
```diff
@@ -2163,6 +2188,45 @@ def _get_disk_size_units(self, image_path):
...
+            msg = ("Error happened when executing command fdisk with"
```
[the remark this answers is not in the document]
Comment (PanYaLian) [reply zvmsdk/smtclient.py]: done

**Comment by jichenjc** [comment]
I don't have further comments, maybe @bjhuangr can take a look

File: zvmsdk/smtclient.py [hunk 6]
```diff
@@ -2201,6 +2226,45 @@ def _get_disk_size_units(self, image_path):
...
         return root_disk_units
 
+    def _get_disk_size_units_rhcos(self, image_path):
+        command = "fdisk -b 4096 -l %s | head -2 | awk '{print $5}'" % (
+                                                                image_path)
+
+        rc = 0
+        output = ""
+        try:
+            output = subprocess.check_output(command, shell=True,
```
Comment (bjhuangr) [inline zvmsdk/smtclient.py]: use zvmskd.utils.execute() to avoid duplicate error handling

**Pull request merged**